mlir/lib/Transforms/Utils/DialectConversion.cpp

```diff
@@ -2527,6 +2527,7 @@ LogicalResult OperationConverter::legalizeConvertedOpResultTypes(
           opReplacement->getConverter());
       rewriterImpl.mapping.map(result, castValue);
       inverseMapping[castValue].push_back(result);
+      llvm::erase(inverseMapping[newValue], result);
```

Comment:
This is probably a very short list. Open to changing it to a Set though.

Reply:
I agree, list is usually very short, so a vector should be fine.